src/user/system/CaptainManager.ts

```diff
@@ -671,6 +679,83 @@ class CaptainManager {
             })
     }
 
+    updateGoAccessInfo(goAccessInfo: GoAccessInfo) {
+        const self = this
+        const dockerApi = this.dockerApi
+        const enabled = goAccessInfo.isEnabled
+
+        // Validate cron schedules
+        if (!Utils.validateCron(goAccessInfo.data.rotationFrequencyCron)) {
+            throw ApiStatusCodes.createError(
+                ApiStatusCodes.ILLEGAL_PARAMETER,
+                'Invalid cron schedule'
+            )
+        }
+
+        const crontabFilePath = `${
+            CaptainConstants.goaccessConfigPathBase
+        }/crontab.txt`
+
+        return Promise.resolve()
+            .then(function () {
+                return self.dataStore.setGoAccessInfo(goAccessInfo)
+            })
+            .then(function () {
+                const cronFile = [
+                    `${goAccessInfo.data.rotationFrequencyCron} /processLogs.sh`,
+                ].join('\n')
+
+                return fs.outputFile(crontabFilePath, cronFile)
+            })
+            .then(function () {
+                return dockerApi.ensureContainerStoppedAndRemoved(
+                    CaptainConstants.goAccessContainerName,
+                    CaptainConstants.captainNetworkName
+                )
+            })
+            .then(function () {
+                if (enabled) {
```

Comment:
We should wait 1000ms or so here. Docker is a bit unreliable in terms of guaranteeing the previous operation to be fully done.